### PR TITLE
feat(s3-proxy): Add `envFrom` to allow multiple ConfigMaps to be passed as environment variables

### PR DESCRIPTION
**Pull request by EternalDeiwos**
The title pretty much describes what this does. Currently the values only allows us to manually specify env values, or pass one existing secret to be mounted into the secrets directory. Ideally we'd be able to mount multiple secrets but this is a decent workaround until something like that exists.

This handles providing both Secrets and ConfigMaps as environment variables which could then be specified for whatever purpose via the config.

Details on `envFrom` can be found [here](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#environment-variables)

This should not introduce any breaking changes.

### CODE REVIEW

File: charts/s3-proxy/templates/deployment.yaml
Comment (oxyno-zeta) [inline charts/s3-proxy/templates/deployment.yaml]: Should be ".Values.envFrom" and I think the indent have to be changed but cannot be sure on phone

**Review by oxyno-zeta — CHANGES_REQUESTED** [review]
Hello @EternalDeiwos !

Thanks for your contribution ! That's really cool !
For the MR in itself: I never used this before but this seems to be a good thing. I will be happy to merge it when you have patched following the comment on the code.

Have a good day!

Oxyno-zeta

File: charts/s3-proxy/templates/deployment.yaml
Comment (EternalDeiwos) [reply charts/s3-proxy/templates/deployment.yaml]: 🤦‍♂️ Fixing.

**Comment by EternalDeiwos** [comment]
@oxyno-zeta I've made the requested changes. Good catch. Any chance you could do a patch release after you've merged this?

**Comment by oxyno-zeta** [comment]
Hello @EternalDeiwos ,

Yes ! The CI will publish automatically the new version. But for this, you have to change the version in the Chart.yaml. New version must be `2.12.0` [here](https://github.com/oxyno-zeta/helm-charts-v2/blob/master/charts/s3-proxy/Chart.yaml#L21) .

Can you edit your MR for that ?

Thanks !

PS: Ignore the CI markdown error.

File: charts/s3-proxy/templates/deployment.yaml
Comment (oxyno-zeta) [reply charts/s3-proxy/templates/deployment.yaml]: No problem :)

**Comment by EternalDeiwos** [comment]
@oxyno-zeta Done

**Review by oxyno-zeta — APPROVED** [review]

**Comment by oxyno-zeta** [comment]
@EternalDeiwos : s3-proxy 2.12.0 chart is released ! Thanks for your contribution !

**Comment by oxyno-zeta** [comment]
Hope you are enjoying the app ;)

**Comment by EternalDeiwos** [comment]
It is certainly useful; thank you for making it.

**Pull request merged**
